### PR TITLE
It should not be possible to assign a volunteer to multiple supervisors at once

### LINKED ISSUE #505: It should not be possible to assign a volunteer to multiple supervisors at once
**Part of epic:** #528 

**Where does/should this occur?**

_Frontend:_ System should reject imports containing multiple `supervisor_volunteer` assignments. (What that would look like in a .csv is multiple supervisor email addresses separated by comma – the system should reject that when a user attempts to import it.)
This should trigger an error message (future work not yet in this issue). 

_Backend:_ 
In the create table `supervisor_volunteer`, it should not be possible for a unique volunteer to have an *active* association with more than one supervisor. (But this should not prevent a volunteer from having many *inactive* associations with more than one supervisor – see issue #449.)

**Screenshots**
This was done via "import supervisor" with file. This should not be possible:
```
email, display_name, supervisor_volunteers
s5@example.com,s5,volunteer1@example.com
s6@example.com,s6,volunteer1@example.com
```

Screen Shot 2020-08-09 at 11 55 44 AM

### PR DESCRIPTION
**Pull request by dominiquecuevas**
### What github issue is this PR for, if any?
Resolves #505

### How will this affect user permissions?
- Volunteer permissions:
- Supervisor permissions:
- Admin permissions:

### Feelings gif (optional)
What gif best describes your feeling working on this issue? https://giphy.com/
How to embed:
`alt text`

### CODE REVIEW

**Comment by compwron** [comment]
Rebase fixed the build! :)

File: app/controllers/supervisor_volunteers_controller.rb [hunk 1]
```diff
@@ -19,7 +19,7 @@ def unassign
     volunteer = Volunteer.find(params[:id])
     supervisor_volunteer = volunteer.supervisor_volunteer
     supervisor_volunteer.is_active = false
-    supervisor_volunteer.save
+    supervisor_volunteer.save!
```
Comment (compwron) [inline app/controllers/supervisor_volunteers_controller.rb:22]: Yay! We might want to add more error handling later but this is much better than failing silently- like it was doing before.

File: app/models/concerns/file_importer.rb [hunk 2]
```diff
@@ -38,7 +38,12 @@ def import_supervisors
           .split(",")
           .map { |email| User.find_by(email: email.strip) }
           .compact
-        user.volunteers << volunteers if volunteers.present?
+
+        volunteers.each do |volunteer|
+          if !volunteer.supervisor
+            user.volunteers << volunteer
```
Comment (compwron) [inline app/models/concerns/file_importer.rb:44]: I want `user` to be renamed to `supervisor` (in a later PR is fine)

File: app/models/concerns/file_importer.rb [hunk 3]
```diff
@@ -68,7 +73,7 @@ def import_cases
     end
     build_message("casa_cases")
   end
-
+  # TODO: add which names were imported when failed imports
```
Comment (compwron) [inline app/models/concerns/file_importer.rb:76]: This TODO still stands- let's do a cleanup PR after this :)

File: app/models/supervisor_volunteer.rb [hunk 4]
```diff
@@ -4,6 +4,12 @@ class SupervisorVolunteer < ApplicationRecord
   belongs_to :volunteer, class_name: "User"
   belongs_to :supervisor, class_name: "User"
   validates :supervisor_id, uniqueness: {scope: :volunteer_id} # only 1 row allowed per supervisor-volunteer pair
+  validates :volunteer_id, uniqueness: {scope: :is_active}, if: :is_active?
+
+  def is_active?
+    is_active == true
```
Comment (compwron) [inline app/models/supervisor_volunteer.rb:10]: I feel like rails gives us something like this for free, let's see if we can simplify :)

File: app/models/supervisor_volunteer.rb [hunk 5]
```diff
@@ -26,4 +32,4 @@ class SupervisorVolunteer < ApplicationRecord
 #
 #  fk_rails_...  (supervisor_id => users.id)
 #  fk_rails_...  (volunteer_id => users.id)
-#
+#
```
Comment (compwron) [inline app/models/supervisor_volunteer.rb:35]: I recommend telling VSCode to not trim line endings- something like this https://stackoverflow.com/questions/39525417/visual-studio-code-how-to-show-line-endings

File: spec/requests/imports_request_spec.rb [hunk 6]
```diff
@@ -60,5 +61,29 @@
 
       expect(response).to redirect_to(imports_url(import_type: "supervisor"))
     end
+    # TODO: Dominique to make test for importing csv for multiple supervisors for 1 volunteer
```
Comment (compwron) [inline spec/requests/imports_request_spec.rb:64]: Is this TODO done? It looks like it

File: spec/requests/supervisor_volunteers_spec.rb [hunk 7]
```diff
@@ -102,7 +101,7 @@
         }.not_to change(supervisor.volunteers, :count)
 
         association.reload
-
+
```
Comment (compwron) [inline spec/requests/supervisor_volunteers_spec.rb:104]: I recommend configuring VSCode to auto-trim trailing spaces on save - probably something like this https://stackoverflow.com/questions/30884131/remove-trailing-spaces-automatically-or-with-a-shortcut

**Review by compwron — APPROVED** [review]
Woot woot woot!!!! ✨✨✨✨

File: app/models/supervisor_volunteer.rb (hunk 4 above)
Comment (dominiquecuevas) [reply app/models/supervisor_volunteer.rb:10]: Hmm, I'm not really sure what you're talking about. 
I tried this `validates :volunteer_id, uniqueness: {scope: :is_active}, if: { :is_active == true }`  and variations of it but it doesn't work, syntax errors. 
Am I on the right track, or are you referring to something else?

Comment (compwron) [reply app/models/supervisor_volunteer.rb:10]: yes that's what I'm thinking! Hmm

File: spec/requests/imports_request_spec.rb (hunk 6 above)
Comment (dominiquecuevas) [reply spec/requests/imports_request_spec.rb:64]: My mistake, yes it's done. I'll remove the line in the next PR

File: app/models/concerns/file_importer.rb (hunk 2 above)
Comment (dominiquecuevas) [reply app/models/concerns/file_importer.rb:44]: I will make a TODO comment, thanks!

File: app/models/supervisor_volunteer.rb (hunk 4 above)
Comment (compwron) [reply app/models/supervisor_volunteer.rb:10]: Note: this was discussed and solved in slack https://rubyforgood.slack.com/archives/CVB0QJGVD/p1599271085161000

**Pull request merged**